test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/JSNativeObjectInPackage.scala

```diff
@@ -0,0 +1,27 @@
+package org.scalajs.testsuite.jsinterop
+
+import org.junit.Test
+import org.junit.Assert._
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation.JSName
+
+package object jsNativeObjectInPackage {
+  @js.native
+  object Foo extends js.Object
+
+  @js.native
+  @JSName("Bar")
+  object Baz extends js.Object
+}
```

Comment:
You don't seem to use `Foo` nor `Baz` in the tests.
Also, you should test classes.